qiita_db/base.py

```diff
@@ -128,14 +131,34 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
         conn_handler = SQLConnectionHandler()
 
         return conn_handler.execute_fetchone(
             "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
             "{0}_id=%s)".format(self._table), (id_, ))[0]
 
+    def _check_portal(self, id_):
+        """Checks that object is accessible in current portal
+
+        Parameters
+        ----------
+        id_ : object
+            The ID to test
+        """
+        if self._portal_table is None:
```

Comment:
Same check that the _check_subclass function is doing actually. It is None by default in the base class, then changed to an actual value in the subclass.